tests/pos/i15872.scala

```diff
@@ -0,0 +1,10 @@
+// From https://github.com/lampepfl/dotty/pull/15872#issuecomment-1218041165
+trait Tag[S]
+case class TupTag[A, T <: Tuple]() extends Tag[A *: T]
+
+def tupleId[T <: Tuple](x: T): x.type = x
+
+def foo[S](x: S, ev: Tag[S]) = ev.match {
+  case _: TupTag[a, t] =>
+    val t1: t = tupleId(x.tail)
+}
```

Comment:
@Linyxus fixed!

Reply:
Nice! 👍

Reply:
One interesting example ([link](https://github.com/lampepfl/dotty/pull/14754/commits/cda22ebe0feac0141dfd249482aeca030bc268d9)) on path-dependent GADT branch fails `-Ycheck` because of MatchType reduction. I think this PR will allow us to bring it back too! :)